### PR TITLE
feat(client): ts-migration for EditorTabs component

### PR DESCRIPTION
**Pull request by Ismailtlem**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Related to #43778

### CODE REVIEW

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Quick linter issue 🙂

**Comment by Ismailtlem** [comment]
@nhcarrigan any feedback left ?

File: client/src/templates/Challenges/classic/editor-tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import { sortChallengeFiles } from '../../../../../utils/sort-challengefiles';
+import { ChallengeFile, ChallengeFiles } from '../../../redux/prop-types';
+import {
+  toggleVisibleEditor,
+  visibleEditorsSelector,
+  challengeFilesSelector
+} from '../redux';
+
+type VisibleEditorsType = {
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/classic/editor-tabs.tsx:14]: Some devs do not like being explicit (👀 @ojeytonwilliams 👀 )

```suggestion
type VisibleEditors = {
```

File: client/src/templates/Challenges/classic/editor-tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+interface EditorTabsProps {
+  challengeFiles: ChallengeFiles;
+  toggleVisibleEditor: typeof toggleVisibleEditor;
+  visibleEditors: VisibleEditorsType;
+}
+
+const mapStateToProps = createSelector(
+  visibleEditorsSelector,
+  challengeFilesSelector,
+  (visibleEditors: VisibleEditorsType, challengeFiles: ChallengeFiles) => ({
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/classic/editor-tabs.tsx:26]:
```suggestion
  (visibleEditors: VisibleEditors, challengeFiles: ChallengeFiles) => ({
```

File: client/src/templates/Challenges/classic/editor-tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  challengeFilesSelector
+} from '../redux';
+
+type VisibleEditorsType = {
+  [key: string]: boolean;
+};
+interface EditorTabsProps {
+  challengeFiles: ChallengeFiles;
+  toggleVisibleEditor: typeof toggleVisibleEditor;
+  visibleEditors: VisibleEditorsType;
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/classic/editor-tabs.tsx:20]:
```suggestion
  visibleEditors: VisibleEditors;
```

File: client/src/templates/Challenges/classic/editor-tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  toggleVisibleEditor
+};
+
+class EditorTabs extends Component<EditorTabsProps> {
+  static displayName: string;
+  render() {
+    const { challengeFiles, toggleVisibleEditor, visibleEditors } = this.props;
+    return (
+      <div className='monaco-editor-tabs'>
+        {/* eslint-disable-next-line @typescript-eslint/no-unsafe-call*/}
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/classic/editor-tabs.tsx:42]:
```suggestion
        {/* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call */}
```

File: client/src/templates/Challenges/classic/editor-tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/classic/editor-tabs.tsx:1]: No need to enable this globally

```suggestion
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you, for another one, Ismail.

Small changes, then this is good to go.

**Review by ShaunSHamilton — APPROVED** [review]
Thank you, for handling that 👍 

This LGTM

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me!

**Pull request merged**